pixeltable/share/protocol/common.py

```diff
@@ -0,0 +1,87 @@
+import uuid
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Optional
+from urllib.parse import urlparse
+
+from pydantic import BaseModel
+
+
+class StorageDestination(str, Enum):
```

Comment:
Can we use the existing StorageTarget enum for this instead of creating something new/duplicative?

Reply:
I'm just moving protocol from cloud to core :)  StorageTarget came after StorageDestination 
every change here we need to change cloud code to reflect it